cmd/common/registry.go

```diff
@@ -0,0 +1,356 @@
+		addFlags(rootCmd, operatorParams)
+	}
+
+	// Add all known gadgets to cobra in their respective categories
+	categories := gadgets.GetCategories()
+	for _, gadget := range gadgetregistry.GetGadgets() {
+		cmd, ok := lookup[gadget.Category()]
+		if !ok {
+			// Category not found, add it
+			categoryDescription, ok := categories[gadget.Category()]
+			if !ok {
+				panic(fmt.Errorf("category unknown: %q", gadget.Category()))
+			}
+			cmd = &cobra.Command{
+				Use:   gadget.Category(),
+				Short: categoryDescription,
+			}
+			rootCmd.AddCommand(cmd)
+			lookup[gadget.Category()] = cmd
+		}
+		cmd.AddCommand(buildCommandFromGadget(gadget, columnFilters, runtime, runtimeParams, operatorsParamsCollection))
+	}
+}
+
+func buildGadgetDoc(gadget gadgets.Gadget, parser parser.Parser) string {
+	var out strings.Builder
+	out.WriteString(gadget.Description() + "\n\n")
+
+	if parser != nil {
+		out.WriteString("Available columns:\n")
+		for columnName, description := range parser.GetColumnNamesAndDescription() {
+			out.WriteString("\t" + columnName + "\n")
+			if description != "" {
+				out.WriteString("\t  " + description + "\n")
+			}
+		}
+	}
+	return out.String()
+}
+
+func buildOutputFormatsHelp(outputFormats gadgets.OutputFormats) []string {
+	var outputFormatsHelp []string
+	var supportedOutputFormats []string
+
+	for ofKey, of := range outputFormats {
+		supportedOutputFormats = append(supportedOutputFormats, ofKey)
+		desc := fmt.Sprintf("%s (%s)", of.Name, ofKey)
+		if of.Description != "" {
+			desc += fmt.Sprintf("\n  %s", of.Description)
+		}
+		outputFormatsHelp = append(outputFormatsHelp, desc)
+	}
+	sort.Strings(outputFormatsHelp)
+	outputFormatsHelp = append([]string{fmt.Sprintf("Output format (%s).", strings.Join(supportedOutputFormats, ", ")), ""}, outputFormatsHelp...)
+	return outputFormatsHelp
+}
+
+func buildCommandFromGadget(gadget gadgets.Gadget,
+	columnFilters []cols.ColumnFilter,
+	runtime runtime.Runtime,
+	runtimeParams *params.Params,
+	operatorsParamsCollection params.Collection,
+) *cobra.Command {
+	var outputMode string
+	var verbose bool
+	var showColumns []string
+	var filters []string
+	var timeout int
+
+	outputFormats := gadgets.OutputFormats{}
+	defaultOutputFormat := ""
+
+	// Instantiate parser - this is important to do, because we might apply filters and such to this instance
+	parser := gadget.Parser()
+	if parser != nil && columnFilters != nil {
+		parser.SetColumnFilters(columnFilters...)
+	}
+
+	// Instantiate gadget params - this is important, because the params get filled out by cobra
+	gadgetParams := gadget.Params().ToParams()
+
+	// Get per gadget operator params
+	validOperators := operators.GetOperatorsForGadget(gadget)
+	operatorsPerGadgetParamCollection := validOperators.PerGadgetParamCollection()
+
+	cmd := &cobra.Command{
+		Use:   gadget.Name(),
+		Short: gadget.Description(),
+		Long:  buildGadgetDoc(gadget, parser),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
nit: Do we we need to set `SilenceUsage` here as well? It seems we are doing it [before](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/cmd/local-gadget/utils/flags.go#L96)?